src/gql/directives/WordsmithTransform.php

```diff
@@ -0,0 +1,341 @@
+<?php
+/**
+ * Wordsmith
+ *
+ * @author     Michael Rog <michael@michaelrog.com>
+ * @link       https://topshelfcraft.com
+ * @copyright  Copyright 2020, Top Shelf Craft (Michael Rog)
+ * @see        https://github.com/topshelfcraft/Wordsmith
+ */
+
+namespace topshelfcraft\wordsmith\gql\directives;
+
+use topshelfcraft\wordsmith\Wordsmith;
+
+use craft\gql\base\Directive;
+use craft\gql\GqlEntityRegistry;
+use GraphQL\Error\Error;
+use GraphQL\Language\DirectiveLocation;
+use GraphQL\Type\Definition\Directive as GqlDirective;
+use GraphQL\Type\Definition\ResolveInfo;
+use GraphQL\Type\Definition\Type;
+
+/**
+ * @author Michael Rog <michael@michaelrog.com>
+ * @package Wordsmith
+ * @since 3.3.0.1
+ */
+class WordsmithTransform extends Directive
+{
+
+
+    /**
+     * @inheritdoc
+     */
+    public static function create(): GqlDirective
+    {
+        if ($type = GqlEntityRegistry::getEntity(self::name())) {
+            return $type;
+        }
+
+        $type = GqlEntityRegistry::createEntity(static::name(), new self([
+            'name' => static::name(),
+            'locations' => [
+                DirectiveLocation::FIELD,
+            ],
+            'description' => 'Run Wordsmith transforms on the value of the field.',
+            'args' => [
+                'function' => [
+                    'name' => 'function',
+                    'type' => Type::string(),
+                    'description' => 'The Wordsmith function to use.'
+                ],
+                // 'args' => [
+                //     'name' => 'args',
+                //     'type' => Type::listOf,
+                //     'description' => 'The Wordsmith function to use.'
+                // ],
+            ],
+        ]));
+
+        return $type;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function name(): string
+    {
+        return 'wordsmith';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function apply($source, $value, array $arguments, ResolveInfo $resolveInfo)
+    {
+        if(isset($arguments['function']))
+        {
+            switch ($arguments['function']) {
```

Comment:
Ideally, we can avoid needing to explicitly map each function by hardcoding. (It adds a maintenance burden if we have to update more than one place every time we add a method.) I think you could use [`method_exists()`](https://www.php.net/manual/en/function.method-exists.php) to "validate" the provided `function` arg, and then [`call_user_func()`](https://www.php.net/manual/en/function.call-user-func.php) to pass in any arguments and get the result.